core/src/main/java/nl/nn/adapterframework/pipes/IteratingPipe.java

```diff
@@ -435,19 +435,20 @@ public MessageOutputStream provideOutputStream(IPipeLineSession session) throws
 		return null; // ancestor MessageSendingPipe forwards provideOutputStream to sender, which is not correct for IteratingPipe
 	}
 
+	@Override
+	public boolean canStreamToNextPipe() {
+		return super.canStreamToNextPipe() && !isCollectResults(); // when collectResults is false, streaming is not necessary or useful
```

Comment:
Performance gezien is het natuurlijk handiger eerst een boolean check te hebben en dan extra logica..